session/session.go

```diff
@@ -1380,6 +1380,7 @@ func (s *session) ShowProcess() util.ProcessInfo {
 	tmp := s.processInfo.Load()
 	if tmp != nil {
 		pi = tmp.(util.ProcessInfo)
+		pi.Mem = s.GetSessionVars().StmtCtx.MemTracker.BytesConsumed()
```

Comment:
Could we do it in `SetProcessInfo`?

Reply:
No we can not. Memory usage is dynamically changed value, we should calculate the memory usage when "show processlist" is called not when a query is being executed.